# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1520548607 18000
#      Thu Mar 08 17:36:47 2018 -0500
# Node ID 1fd64fa2cebc7edaf35abdf15829cf61f555be19
# Parent  96c74e33d17a4920415d9614e676ed378728bc75
# Parent  dfc017e465f236c4f74577ba21af8843776c7c18
maint: merge stable to default

diff --git a/libinterp/operators/op-int.h b/libinterp/operators/op-int.h
--- a/libinterp/operators/op-int.h
+++ b/libinterp/operators/op-int.h
@@ -345,17 +345,17 @@ along with Octave; see the file COPYING.
   DEFBINOP (PFX ## _ldiv, TS ## scalar, TM ## matrix)                   \
   {                                                                     \
    const octave_ ## TS ## scalar& v1 = dynamic_cast<const octave_ ## TS ## scalar&> (a1); \
    const octave_ ## TM ## matrix& v2 = dynamic_cast<const octave_ ## TM ## matrix&> (a2); \
                                                                         \
    if (! v1.TS ## scalar_value ())                                      \
      warn_divide_by_zero ();                                            \
                                                                         \
-   octave_value retval = octave_value (v2.TS ## scalar_value () / v1.TS ## scalar_value ()); \
+   octave_value retval = octave_value (v2.TM ## array_value () / v1.TS ## scalar_value ()); \
    return retval;                                                       \
    }                                                                    \
                                                                         \
   DEFINTNDBINOP_OP (PFX ## _el_mul, TS ## scalar, TM ## matrix, TS ## scalar, TM ## array, *, TI) \
   DEFBINOP (PFX ## _el_div, TS ## scalar, TM ## matrix)                 \
   {                                                                     \
    const octave_ ## TS ## scalar& v1 = dynamic_cast<const octave_ ## TS ## scalar&> (a1); \
    const octave_ ## TM ## matrix& v2 = dynamic_cast<const octave_ ## TM ## matrix&> (a2); \
